### DISCUSSION TITLE
Checking if there are changes to be executed BEFORE their execution

### DISCUSSION THREAD

**Discussion by igorg-il** [flamingock/mongock#16] [Q&A]
In some stares of our application we want to prevent mongock change sets from being executed (e.g. writing error log or preventing app from start)
Is there a way to check if there is any change to execute BEFORE we actually execute them?
In class MigrationExecutionBase there is private method isThereAnyChangeSetItemToBeExecuted which does the job but it is internal.

**Comment by osantana85** [comment]
Hello @igorg-il , could you explain your use case in more detail? Are you expecting a mechanism to ask Mongock if there are any changes to be executed and, if so, not execute them, or even not start the application?

**Reply by igorg-il** [reply]
@osantana85 right
Not execute change sets and not start the application
The reason is that  customer can move application to edit mode in which case he can change configuration by himself which causes changes in db. In such case we don't want mongock to override hiss changes.
